src/linear_algebra/basic.lean

```diff
@@ -497,9 +497,6 @@ instance : has_top (submodule R M) :=
 
 @[simp] lemma mem_top : x ∈ (⊤ : submodule R M) := trivial
 
-lemma eq_bot_of_zero_eq_one (zero_eq_one : (0 : R) = 1) : p = ⊥ :=
-by ext x; simp [semimodule.eq_zero_of_zero_eq_one x zero_eq_one]
```

Comment:
This one now falls out of `submodule.unique'` once you convert `zero_eq_one` to `subsingleton R`.